lib/Github/HttpClient/Plugin/History.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Github\HttpClient\Plugin;
+
+use Http\Client\Common\Plugin\Journal;
+use Http\Client\Exception;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A plugin to remember the last response.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class History implements Journal
+{
+    /**
+     * @var ResponseInterface
+     */
+    private $lastResponse;
+
+    /**
+     * @return ResponseInterface|null
+     */
+    public function getLastResponse()
+    {
+        return $this->lastResponse;
+    }
+
+    public function addSuccess(RequestInterface $request, ResponseInterface $response)
+    {
+        $this->lastResponse = $response;
+    }
+
+    public function addFailure(RequestInterface $request, Exception $exception)
+    {
```

Comment:
This is not a failed response, it is a failed **request**. When a request throws an exception. 

A "failure" is not when the sever answers with 4xx or 5xx.